src/app/clusters/level-control/level-control.cpp

```diff
@@ -970,18 +977,48 @@ static void moveHandler(app::CommandHandler * commandObj, const app::ConcreteCom
         goto send_default_response;
     }
 
-    if (!rate.IsNull() && (rate.Value() == 0))
+    uint8_t eventDuration; // use this local var so state->eventDurationMs is only set once the command is validated.
+#ifndef IGNORE_LEVEL_CONTROL_CLUSTER_TRANSITION
+    // If the Rate field is null, the device should move at the default move rate, if available,
+    // Otherwise, move as fast as possible
+    if (rate.IsNull())
     {
-        // Move at a rate of zero is no move at all. Immediately succeed without touching anything.
-        ChipLogProgress(Zcl, "Immediate success due to move rate of 0 (would move at no rate).");
-        status = Status::Success;
-        goto send_default_response;
+        app::DataModel::Nullable<uint8_t> defaultMoveRate;
+        status = Attributes::DefaultMoveRate::Get(endpoint, defaultMoveRate);
+        if (status != Status::Success || defaultMoveRate.IsNull())
+        {
+            ChipLogProgress(Zcl, "ERR: reading default move rate %x", to_underlying(status));
+            eventDuration = FASTEST_TRANSITION_TIME_MS;
+        }
+        else
+        {
+            // Error out if DefaultMoveRate is equal to 0
+            if (defaultMoveRate.Value() == 0)
+            {
+                status = Status::InvalidCommand;
```

Comment:
This is not ideal, since it claims something is wrong with the command that was sent, but something is actually wrong with the stored state of the device.

Ideally the spec would define the behavior here.  Until it does, I'd suggest a more relevant error code might be InvalidInState or Failure or something.